### PR TITLE
Chore: more typing improvements to @grafana/ui

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- removing `any`/type assertions from `@grafana/ui`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-ui/src/components/Logs/LogRowContextProvider.tsx [hunk 1]
```diff
@@ -151,7 +151,7 @@ export const LogRowContextProvider: React.FunctionComponent<LogRowContextProvide
   // React Hook that creates an object state value called result to component state and a setter function called setResult
   // The initial value for result is null
   // Used for sorting the response from backend
-  const [result, setResult] = useState<ResultType>(null as any as ResultType);
+  const [result, setResult] = useState<ResultType>(null as unknown as ResultType);
```
Comment (matyax) [inline packages/grafana-ui/src/components/Logs/LogRowContextProvider.tsx:154]:
```suggestion
  const [result, setResult] = useState<ResultType | null>(null);
```

**Review by matyax — COMMENTED** [review]
Great improvements! Left a couple of comments.

File: packages/grafana-ui/src/components/Logs/LogRowContextProvider.tsx (hunk 1 above)
Comment (ashharrison90) [reply packages/grafana-ui/src/components/Logs/LogRowContextProvider.tsx]: ah thanks, this is much nicer 😍 also exposed a problem in the useEffect which i've now fixed as well 👍

File: packages/grafana-ui/src/components/QueryField/QueryField.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ export interface QueryFieldProps extends Themeable2 {
   onBlur?: () => void;
   onChange?: (value: string) => void;
   onRichValueChange?: (value: Value) => void;
-  onClick?: (event: Event | React.MouseEvent, editor: Editor, next: () => any) => any;
+  onClick?: EventHook<React.MouseEvent<Element, MouseEvent>>;
```
[the remark this answers is not in the document]
Comment (ashharrison90) [reply packages/grafana-ui/src/components/QueryField/QueryField.tsx:45]: the new `EventHook` type is exactly the same as the type we'd defined there (still includes the `editor`/`next` args), i've just imported it from the `slate-react` package to save us from redefining it in our code.

**Review by matyax — APPROVED** [review]
👏 👏

**Pull request merged**